pkg/providers/launchtemplate/suite_test.go

```diff
@@ -548,6 +548,16 @@ var _ = Describe("LaunchTemplates", func() {
 			ExpectProvisioned(ctx, env.Client, cluster, cloudProvider, prov, pod)
 			ExpectNotScheduled(ctx, env.Client, pod)
 		})
+		It("should pack pods if the sum of pod ephemeral-storage and overhead exceeds node capacity and instance storage is mounted", func() {
+			nodeClass.Spec.InstanceStorePolicy = lo.ToPtr(v1beta1.InstanceStorePolicyRAID0)
+			ExpectApplied(ctx, env.Client, nodePool, nodeClass)
+			pod := coretest.UnschedulablePod(coretest.PodOptions{ResourceRequirements: v1.ResourceRequirements{
+				Requests: map[v1.ResourceName]resource.Quantity{
+					v1.ResourceEphemeralStorage: resource.MustParse("19Gi"),
+				}}})
+			ExpectProvisioned(ctx, env.Client, cluster, cloudProvider, prov, pod)
+			ExpectScheduled(ctx, env.Client, pod)
```

Comment:
Worth checking the instance type that this scheduled to to ensure that it's an instance store instance type?

Reply:
Makes sense to me, added check for the instance-store instance & ephemeral-storage capacity in https://github.com/aws/karpenter-provider-aws/pull/4735/commits/4bf56c65efa9007b3aee0871e6d554e370fa549b

Reply:
actually this explicitly checks `m6idn.32xlarge`, can change that to be more dynamic if preferred

Reply:
Sounds good 👍🏼 The current state of the code lgtm